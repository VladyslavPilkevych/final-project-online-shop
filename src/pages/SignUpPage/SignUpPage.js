/* eslint-disable max-len */
import React from 'react';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { useNavigate, Link } from 'react-router-dom';
import axios from 'axios';
import CustomInput from '../../components/CustomInput/CustomInput';
import { createNewCustomer } from '../../api/api';
import styles from './SignUpPage.module.scss';

function SignUpPage() {
  const navigate = useNavigate();

  const initialValues = {
    firstName: '',
    lastName: '',
    login: '',
    email: '',
    password: '',
    telephone: '',
    gender: '',
    avatarUrl: '../../assets/icons/defaultUserIcon.png',
    isAdmin: false,
  };

  const onSubmit = async (values, { resetForm }) => {
    if (values.password !== values.repeatPassword) {
      alert('Passwords should be equal');
    }
    const newCustomer = values;
    const response = await createNewCustomer(newCustomer).then((user) => user).catch((err) => console.log(err));

    if (response && response.status === 200) {
      resetForm();
      navigate({ pathname: '/sign-in' });
    }
    console.log(response);
  };

  const validationSchema = yup.object().shape({
    firstName: yup
      .string()
      .required('Поле обязательно')
      .matches(/[A-Za-z ]/gi, 'Только латинские'),
    lastName: yup
      .string()
      .required('Поле обязательно')
      .matches(/[A-Za-z ]/gi, 'Только латинские'),
    login: yup
      .string()
      .required('Поле обязательно')
      .matches(/[A-Za-z ]/gi, 'Только латинские'),
    // email: yup.string().email('Неверный email').required('Поле обязательно'),
    // password: yup.string().required('Поле обязательно'),
    // repeatPassword: yup.string().required('Поле обязательно'),
  });

  return (
    <div className={styles.signUpPageWrapper}>
      <h1 className={styles.mainTitle}>Customer registration</h1>
      <div className={styles.signUpBlocksPageWrapper}>
        <div className={styles.signUpBlockWrapper}>
          <h2 className={styles.customersBlocksTitle}>Customers information</h2>
          <p className={styles.customersBlocksSubtitle}>If you are not have an account, fill information below</p>
          <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={validationSchema}>
            <Form>
              <CustomInput name="firstName" type="text" placeholder="FirstName" />
              <CustomInput name="lastName" type="text" placeholder="LastName" />
              <CustomInput name="login" type="text" placeholder="Login" />
              <CustomInput name="email" type="text" placeholder="Email" />
              <CustomInput name="telephone" type="tel" placeholder="Telephone" />
              <CustomInput name="gender" type="text" placeholder="Gender" />
              {/* <CustomInput
                name="avatarUrl"
                type={"url" || "file"}
                placeholder="avatarUrl"
              /> */}
              <CustomInput name="password" type="password" placeholder="Password" />
              <CustomInput name="repeatPassword" type="password" placeholder="Repeat password" />

              <button type="submit" className={styles.createCustomerButton}>
                Create account
              </button>
            </Form>
          </Formik>
        </div>
        <div className={`${styles.signUpBlockWrapper} ${styles.signInBlockWrapper}`}>
          <h2 className={styles.customersBlocksTitle}>Already have registered Account?</h2>
          <p className={styles.customersBlocksSubtitle}>If you are not have an account, you can Login on button below</p>
          <Link to="/sign-in">
            <button type="button" className={styles.createCustomerButton}>
              Log In Page
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export default SignUpPage;
