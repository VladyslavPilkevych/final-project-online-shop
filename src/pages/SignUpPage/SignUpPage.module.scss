@import '../../assets/styles/global.module.scss';

.signUpPageWrapper {
    max-width: 1398px;
    margin: 0 auto;
}

.mainTitle {
    font-size: 32px;
    font-weight: bold;
    font-family: $fonts;
    margin-top: 58px;
}

.signUpBlocksPageWrapper {
    display: flex;
    column-gap: 34px;
    justify-content: center;
    margin-top: 21px;
}

.signUpBlockWrapper {
    background-color: $bg-white-color;
    width: 35%;
    padding: 37px 60px 25px 60px;
}

.customersBlocksTitle {
    font-size: 18px;
    font-weight: 600;
    line-height: 27px;
}

.customersBlocksSubtitle {
    font-size: 14px;
    font-family: $fonts;
    font-style: normal;
    font-weight: 300;
    line-height: 30px;
}

.createCustomerButton {
    margin-top: 30px;
    background-color: #2e58ff;
    color: $white;
    border-radius: 50px;
    padding: 10px;
    position: relative;
	display: inline-block;
    transition: 0.5s;
    border: none;
    // border: 5px solid $bg-white-color;
}
.createCustomerButton:hover {
	background-color: darken(#1d4aff, 5%);
	box-shadow: 2px 2px 15px 5px #7b6aff;
	color: $white;
	cursor: pointer;
}
