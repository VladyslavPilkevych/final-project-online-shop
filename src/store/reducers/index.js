import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
