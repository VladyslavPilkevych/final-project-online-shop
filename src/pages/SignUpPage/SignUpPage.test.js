import { render } from '@testing-library/react';
import SignUpPage from './SignUpPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <SignUpPage />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot SignUpPage should render', () => {
  test('Snapshot SignUpPage is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});