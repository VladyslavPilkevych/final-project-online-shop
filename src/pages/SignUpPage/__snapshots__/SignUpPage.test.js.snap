// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot SignUpPage should render Snapshot SignUpPage is rendered 1`] = `
<DocumentFragment>
  <div
    class="signUpPageWrapper"
  >
    <h1
      class="mainTitle"
    >
      Customer registration
    </h1>
    <div
      class="signUpBlocksPageWrapper"
    >
      <div
        class="signUpBlockWrapper"
      >
        <h2
          class="customersBlocksTitle"
        >
          Customers information
        </h2>
        <p
          class="customersBlocksSubtitle"
        >
          If you are not have an account, fill information below
        </p>
        <form
          action="#"
        >
          <p
            class="inputLabel"
          >
            FirstName*
          </p>
          <input
            class="customInput"
            name="firstName"
            placeholder="FirstName"
            type="text"
            value=""
          />
          <p
            class="inputLabel"
          >
            LastName*
          </p>
          <input
            class="customInput"
            name="lastName"
            placeholder="LastName"
            type="text"
            value=""
          />
          <p
            class="inputLabel"
          >
            Login*
          </p>
          <input
            class="customInput"
            name="login"
            placeholder="Login"
            type="text"
            value=""
          />
          <p
            class="inputLabel"
          >
            Email*
          </p>
          <input
            class="customInput"
            name="email"
            placeholder="Email"
            type="text"
            value=""
          />
          <p
            class="inputLabel"
          >
            Phone*
          </p>
          <input
            class="customInput"
            name="telephone"
            placeholder="Phone"
            type="tel"
            value=""
          />
          <p
            class="inputLabel"
          >
            Gender
          </p>
          <input
            class="customInput"
            name="gender"
            placeholder="Gender"
            type="text"
            value=""
          />
          <p
            class="inputLabel"
          >
            Password*
          </p>
          <input
            class="customInput"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <p
            class="inputLabel"
          >
            Repeat password*
          </p>
          <input
            class="customInput"
            name="repeatPassword"
            placeholder="Repeat password"
            type="password"
            value=""
          />
          <button
            class="createCustomerButton"
            type="submit"
          >
            Create account
          </button>
        </form>
      </div>
      <div
        class="signUpBlockWrapper signInBlockWrapper"
      >
        <h2
          class="customersBlocksTitle"
        >
          Already have registered Account?
        </h2>
        <p
          class="customersBlocksSubtitle"
        >
          If you are not have an account, you can Login on button below
        </p>
        <a
          href="/sign-in"
        >
          <button
            class="createCustomerButton"
            type="button"
          >
            Log In Page
          </button>
        </a>
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
